### ISSUE TITLE
Windows OpenSSH error on version 2.10

### ISSUE THREAD

**Issue by danilofuchs** [paramiko/paramiko#2008]
Upgrading from 2.9.2 to 2.10.2 caused issues on Windows.

Command:
```python
ssh_client = paramiko.SSHClient()
ssh_client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
ssh_client.connect(self.ip, username=self.username, password=self.password, timeout=self.timeout_s)

data = 'cat some_file.txt'
ssh_client.exec_command(data, get_pty=True)
```

## 2.9.2:

Runs normally

## 2.10.2:

```
[Errno 22] Invalid argument: '\\\\.\\pipe\\openssh-ssh-agent'
```

**Comment by Daviid-P** [comment]
I'm having the same problem.

```py
from pathlib import Path
PIPE_NAME = Path(r"\\.\pipe\openssh-ssh-agent") 
# WindowsPath(PIPE_NAME)
PIPE_NAME.exists() 
# True
_pipe = open(PIPE_NAME, "rb+", buffering=0)     
_pipe
# <_io.FileIO name='\\\\.\\pipe\\openssh-ssh-agent' mode='rb+' closefd=True>
```

Manually it doesn't seem to fail.

Using Python 3.10.2

**Comment by jun66j5** [comment]
I found https://stackoverflow.com/a/67102903.

> os.path.exists delegates to os.stat, and on Windows, os.stat tries to [open](https://github.com/python/cpython/blob/v3.9.4/Modules/posixmodule.c#L1826) the file:
> ...
> Then it closes the file. At this point, the client can't reopen the pipe unless the server side first calls [DisconnectNamedPipe](https://docs.microsoft.com/en-us/windows/win32/api/namedpipeapi/nf-namedpipeapi-disconnectnamedpipe), which doesn't happen in this code. Also, when the server side tries to read from the pipe, it's reading from os.stat's connection attempt, which closed the pipe without writing any data.

That is reproduce like this:
```
>>> PIPE_NAME = r"\\.\pipe\openssh-ssh-agent"
>>> os.path.exists(PIPE_NAME) and open(PIPE_NAME, "rb+", buffering=0)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
OSError: [Errno 22] Invalid argument: '\\\\.\\pipe\\openssh-ssh-agent'
```
Or:
```
>>> [open(PIPE_NAME, "rb+", buffering=0) for i in range(1)]
[<_io.FileIO name='\\\\.\\pipe\\openssh-ssh-agent' mode='rb+' closefd=True>]
>>> [open(PIPE_NAME, "rb+", buffering=0) for i in range(2)]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 1, in <listcomp>
OSError: [Errno 22] Invalid argument: '\\\\.\\pipe\\openssh-ssh-agent'
```

**Comment by jun66j5** [comment]
Could you please try the following patch (using `os.listdir()` insstead of `os.path.exists()`)?

```diff
--- paramiko/win_openssh.py.orig	2022-03-17 12:59:24.943534700 +0900
+++ paramiko/win_openssh.py	2022-03-17 13:17:06.879519400 +0900
@@ -23,7 +23,9 @@
 
 
 def can_talk_to_agent():
-    return os.path.exists(PIPE_NAME)
+    dir_, name = os.path.split(PIPE_NAME)
+    name = name.lower()
+    return any(name == n.lower() for n in os.listdir(dir_))
 
 
 class OpenSSHAgentConnection:
```

**Comment by jun66j5** [comment]
I just posted #2010. I've tested with Python 2.7 and 3.10 on Windows 10. Could you please try the changes?

**Comment by AdamPelc** [comment]
I had the same problem and with patch(#2010) applied it's no longer an issue.

**Comment by keslerm** [comment]
Hitting this same issue

**Comment by atbigelow** [comment]
Same issue and the changes in #2010 also fixed it for me.

**Comment by bitprophet** [comment]
#2010 merged, will be out in next 2.10.x bugfix.
